File: Python/Krafton_Jungle_1week/assert_example.py
def divide(numerator, denominator):
    # 입력값이 정수가 아니라면 에러를 발생시킨다.
    assert type(numerator) == int, "Numerator must be an integer"
    assert type(denominator) == int, "Denominator must be an integer"
    # 분모가 0이라면 에러를 발생시킨다.
    assert denominator != 0, "Denominator cannot be zero"
    return numerator/denominator

print(divide(10, 0))
    

File: Python/Krafton_Jungle_1week/assert_example.cpp
#include <iostream>
#include <cassert>
int divide(int numerator, int denominator) {
	// 분모가 0이 아니면 에러 발생
	assert(denominator != 0);
	return (double)numerator/denominator;
}

int main() {
	std::cout << divide(10, 0) << std::endl;
	return 0;
}